test/e2e/scenarios/pod_selector.go

```diff
@@ -2,24 +2,106 @@ package scenarios
 
 import (
 	. "github.com/onsi/ginkgo"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	framework "github.com/submariner-io/submariner/test/e2e/framework"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+var f = framework.NewDefaultFramework("podselectornp")
+
+var pod_selector_np = &networkingv1.NetworkPolicy {
+	ObjectMeta: metav1.ObjectMeta {
+		Name: "pod1-pod2-same-ns-different-labels",
+		Namespace: "boring",
+	},
+	Spec: networkingv1.NetworkPolicySpec {
+		PodSelector: metav1.LabelSelector {
+			MatchLabels: map[string]string {
+				"role": "backend",
+			},
+		},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule {{
+			From: []networkingv1.NetworkPolicyPeer {{
+				PodSelector: &metav1.LabelSelector {
+					MatchLabels: map[string]string {
+						"role": "frontend",
+					},
+				},
+			}},
+			Ports: []networkingv1.NetworkPolicyPort {{
+				//Protocol: TCP is the default protocol. Need not mention it
+				Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 1234},
+				// '1234' value comes from the https://github.com/submariner-io/submariner/blob/master/test/e2e/framework/network_pods.go#L13
```

Comment:
Yeah, I added a "TestPort" constant in the refactor review, we can use that constant later.